### PR TITLE
Make flake8 its own CI job

### PR DESCRIPTION
**Pull request by nmuesch**
Change the display name of the azure pipeline job to be the TOKENV being run

### CODE REVIEW

File: .azure-pipelines/unit_tests.yml [hunk 1]
```diff
@@ -36,6 +36,9 @@ jobs:
       Pypy3:
         PYTHON_VERSION: "pypy3"
         TOXENV: "pypy3"
+      Lint:
+        PYTHON_VERSION: "3.5"
+        TOXENV: "flake8"
```
Comment (zippolyte) [inline .azure-pipelines/unit_tests.yml]: Let's also remove it from the Py35 env then

Comment (nmuesch) [reply .azure-pipelines/unit_tests.yml]: Done

**Pull request merged**